echopype/convert/set_groups_azfp.py

```diff
@@ -208,7 +212,7 @@ def set_platform(self) -> xr.Dataset:
                 ),
                 "vertical_offset": (
                     ["time2"],
-                    [np.nan] * len(time2),
+                    depth,
                     self._varattrs["platform_var_default"]["vertical_offset"],
                 ),
                 "water_level": (
```

Comment:
`water_level` should be set to 0.0 (a scalar value) if pressure is present. If pressure is not present, it stays as it is now, `np.nan`. See https://github.com/OSOceanAcoustics/echopype/issues/1181#issue-1910068233

Reply:
Should this be if `pressure` value is equal to `np.nan` then `water_level` should also be `np.nan`? Since, if  `pressure` value cannot be calculated due to required parameters not being present in the XML file then it would be set to `np.nan`.

Reply:
> Should this be if `pressure` value is equal to `np.nan` then `water_level` should also be `np.nan`? 

Correct. Then, if pressure values are not all nan (there are valid values), `water_level` should be set to 0.0, a scalar. `water_level` is a variable that is separate but related to `vertical_offset` (see [here](https://echopype.readthedocs.io/en/stable/data-proc-additional.html#vertical-coordinate-z-axis-variables)). For AZFP, we're allowing `water_level` to only be either `np.nan` or `0.0`.

Reply:
Updated!